src/includes/product.ts

```diff
@@ -16,6 +16,14 @@ export interface Product extends Record<string, any> {
   variants?: ProductVariants;
 }
 
+/**
+ * The type of a function used to transform record data into products.
+ *
+ * @typeparam T The type of the product into which the Record will be
+ * transformed.
+*/
```

Comment:
Indentation. Same goes for the rest of the doc comments introduced in this patch.

```suggestion
  */
```